coderd/promoauth/oauth2.go

```diff
@@ -58,6 +71,53 @@ func NewFactory(registry prometheus.Registerer) *Factory {
 				"source",
 				"status_code",
 			}),
+			rateLimit: factory.NewGaugeVec(prometheus.GaugeOpts{
+				Namespace: "coderd",
+				Subsystem: "oauth2",
+				Name:      "external_requests_rate_limit_total",
+				Help:      "The total number of allowed requests per interval.",
+			}, []string{
+				"name",
+				// Resource allows different rate limits for the same oauth2 provider.
+				// Some IDPs have different buckets for different rate limits.
+				"resource",
+			}),
+			rateLimitRemaining: factory.NewGaugeVec(prometheus.GaugeOpts{
+				Namespace: "coderd",
+				Subsystem: "oauth2",
+				Name:      "external_requests_rate_limit_remaining",
+				Help:      "The remaining number of allowed requests in this interval.",
+			}, []string{
+				"name",
+				"resource",
+			}),
+			rateLimitUsed: factory.NewGaugeVec(prometheus.GaugeOpts{
+				Namespace: "coderd",
+				Subsystem: "oauth2",
+				Name:      "external_requests_rate_limit_used",
+				Help:      "The number of requests made in this interval.",
+			}, []string{
+				"name",
+				"resource",
+			}),
+			rateLimitReset: factory.NewGaugeVec(prometheus.GaugeOpts{
+				Namespace: "coderd",
+				Subsystem: "oauth2",
+				Name:      "external_requests_rate_limit_next_reset_unix",
+				Help:      "Unix timestamp of the next interval",
```

Comment:
Minor clarification, discard if you feel it's muddier.

```suggestion
				Help:      "Unix timestamp for when the next interval starts",
```